test/webhook-apicoverage/resourcetree/updateCoverage_test.go

```diff
@@ -29,42 +29,39 @@ func TestSimpleStructValue(t *testing.T) {
 	}
 }
 
-func TestPtrValueAllCovered(t *testing.T) {
-	tree := getTestTree(ptrTypeName, reflect.TypeOf(ptrType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getPtrTypeValueAllCovered()))
-	if err := verifyPtrValueAllCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestPtrValueSomeCovered(t *testing.T) {
-	tree := getTestTree(ptrTypeName, reflect.TypeOf(ptrType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getPtrTypeValueSomeCovered()))
-	if err := verifyPtrValueSomeCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestArrValueAllCovered(t *testing.T) {
-	tree := getTestTree(arrayTypeName, reflect.TypeOf(arrayType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getArrValueAllCovered()))
-	if err := verifyArryValueAllCovered(tree.Root); err != nil {
-		t.Fatal(err)
+func TestUpdateCoverage(t *testing.T) {
+	datas := []struct {
+		TestName string
+		name     string
+		typeI    interface{}
+		value    interface{}
+		f        func(NodeInterface) error
+	}{
+		{
```

Comment:
nit: use `{{`, `}, {` and `}}` to reduce whitespace.

Reply:
done